src/libLookup/Lookup.cpp

```diff
@@ -1642,17 +1708,417 @@ bool Lookup::InitMining()
+                      "ProcessSetOfflineLookups recvd offline lookup "
+                          << i << ": " << peer);
+        }
     }
 
+    {
+        unique_lock<mutex> lock(m_mutexOfflineLookupsUpdation);
+        m_fetchedOfflineLookups = true;
+        m_offlineLookupsCondition.notify_one();
+    }
+#endif // IS_LOOKUP_NODE
+    return true;
+}
+
+#ifdef IS_LOOKUP_NODE
+void Lookup::StartSynchronization()
+{
+    LOG_MARKER();
+
+    this->CleanVariables();
+
+    auto func = [this]() -> void {
+        GetMyLookupOffline();
+        GetDSInfoFromLookupNodes();
+        while (m_syncType != SyncType::NO_SYNC)
+        {
+            GetDSBlockFromLookupNodes(m_mediator.m_dsBlockChain.GetBlockCount(),
+                                      0);
+            GetTxBlockFromLookupNodes(m_mediator.m_txBlockChain.GetBlockCount(),
+                                      0);
+            this_thread::sleep_for(chrono::seconds(NEW_NODE_SYNC_INTERVAL));
+        }
+    };
+    DetachedFunction(1, func);
+}
+
+Peer Lookup::GetLookupPeerToRsync()
+{
+    LOG_MARKER();
+
+    std::vector<Peer> t_Peers;
+    for (auto p : m_lookupNodes)
+    {
+        if (p == m_mediator.m_selfPeer)
+        {
+            // SKIP
+        }
+        else
+        {
+            t_Peers.push_back(p);
+        }
+    }
+
+    int index = rand() % t_Peers.size();
+
+    return t_Peers[index];
+}
+
+std::vector<unsigned char> Lookup::ComposeGetLookupOfflineMessage()
+{
+    LOG_MARKER();
+
+    // getLookupOfflineMessage = [Port]
+    vector<unsigned char> getLookupOfflineMessage
+        = {MessageType::LOOKUP, LookupInstructionType::SETLOOKUPOFFLINE};
+    unsigned int curr_offset = MessageOffset::BODY;
+
+    Serializable::SetNumber<uint32_t>(getLookupOfflineMessage, curr_offset,
+                                      m_mediator.m_selfPeer.m_listenPortHost,
+                                      sizeof(uint32_t));
+    curr_offset += sizeof(uint32_t);
+
+    return getLookupOfflineMessage;
+}
+
+std::vector<unsigned char> Lookup::ComposeGetLookupOnlineMessage()
+{
+    LOG_MARKER();
+
+    // getLookupOnlineMessage = [Port]
+    vector<unsigned char> getLookupOnlineMessage
+        = {MessageType::LOOKUP, LookupInstructionType::SETLOOKUPONLINE};
+    unsigned int curr_offset = MessageOffset::BODY;
+
+    Serializable::SetNumber<uint32_t>(getLookupOnlineMessage, curr_offset,
+                                      m_mediator.m_selfPeer.m_listenPortHost,
+                                      sizeof(uint32_t));
+    curr_offset += sizeof(uint32_t);
+
+    return getLookupOnlineMessage;
+}
+
+bool Lookup::GetMyLookupOffline()
+{
+    LOG_MARKER();
+
+    // Remove selfPeerInfo from m_lookupNodes
+    auto iter = std::find(m_lookupNodes.begin(), m_lookupNodes.end(),
+                          m_mediator.m_selfPeer);
+    if (iter != m_lookupNodes.end())
+    {
+        m_lookupNodesOffline.push_back(*iter);
```

Comment:
can we not use ptr here? can we just push in m_mediator.m_selfPeer

Reply:
Need to make sure the m_lookupNodesOffline and m_lookupNodes must have one and only one between them two

Reply:
there should only be one copy right?

Reply:
yes, and should have and must have only one